src/Nova/Role.php

```diff
@@ -88,11 +97,17 @@ public function fields(Request $request)
                         'label'  => __($permission->name),
                     ];
                 })
-                ->groupBy('group')
-                ->toArray()),
+                    ->groupBy('group')
+                    ->toArray()),
 
             Text::make(__('Users'), function () {
-                return $this->users()->count();
+                // For the detail page, we have no eager load $with, so, in order
```

Comment:
Why it's necessary load `users` relations to count?

the orginal source `return $this->users()->count()` avoid it.

Reply:
This is also causing an N+1 problem. Calling `count()` on the relation method `users()` will constantly force a query. 

So, on the index page, if we have 25 Roles, we will have 25 unnecessary queries.

Reply:
I think I understood what you meant. Let me make a little adjustment then.

Reply:
Can you please take a look now?